### DISCUSSION TITLE
Bidirectional charging

### DISCUSSION THREAD

**Discussion by sameerpurwar** [zach401/acnportal#114] [General]
I am looking into viable use case for bidirectional charging in the context of V2B, and whether it can help to shave off peak loads of the building energy consumption while maintaining the users requests as well.
Few questions I have :

Can I use bidirectional charging with the ACN framework for my study ?
Does it make sense to have make use of bi-directional charging as it leads to battery degradation as well ?
Is it a viable business use case ?

Thanks in advance

**Comment by sunash** [comment]
Hi Sameer, thanks for your question

You can technically construct a new EVSE that allows for a negative charging current. However, the ACN framework does not implement battery models for bidirectional charging, so results from doing this might not be physically accurate. As an initial exploration, it should be fine to directly extend the EVSE class, but a full study would likely need further consideration.

As for your other two questions, I'm not well-versed in the V2G/V2B research, but they are hot topics in academia right now, with some traction in industry as well. I assume they've thought about degradation and economic viability, so I'd recommend seeking out some literature on the subject. Let me know if you have further questions!
